### DISCUSSION TITLE
Connecting to an LSP server via an SSH tunnel

### DISCUSSION THREAD

**Discussion by duncan-bayne** [joaotavora/eglot#1463] [General]
Hi :)

First up, thanks for eglot :heart:

I'm using Emacs to edit files "locally", that are in fact on an NFS share on a VM, mounted into my local filesystem.

To support this I'd like to connect eglot to an LSP server (Solargraph, in this case) running on the same VM.  It's easy enough to start it on the VM, then create an SSH tunnel on port 7658.

But I can't seem to find any way of having eglot _not_ start a new server, but just connect to an existing one via the tunnel.

Am I missing some documentation somewhere, or is this not currently possible?

Yours,
Duncan

**Comment by joaotavora** [comment]
It _should_ be possible, see the [manual](https://joaotavora.github.io/eglot/#Setting-Up-LSP-Servers) and soon you'll find hte "host/port" section.  But it's rarely tested.

**Comment by duncan-bayne** [comment]
Thanks @joaotavora!  I did look at the docs but must have missed that.  I'll give it a whirl and report back here.

**Comment by duncan-bayne** [comment]
Yup, that worked a treat :grin: 

```
(add-to-list 'eglot-server-programs
               '(ruby-mode . ("localhost" 7658)))
```
